src/test/java/lotto/domain/LottoNumbersTest.java

```diff
@@ -0,0 +1,42 @@
+package lotto.domain;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatIllegalArgumentException;
+
+
+class LottoNumbersTest {
+
+    static Stream<Arguments> generateLottoNumbers() {
+        return Stream.of(
+                Arguments.of(Arrays.asList(1, 2, 3, 4, 5, 6), 6),
+                Arguments.of(Arrays.asList(7, 8, 9, 10, 11, 12), 0)
+        );
+    }
+
+    @DisplayName("인자로 받은 로또 번호 크기가 6개가 아니면 IllegalArgumentException을 발생시킨다.")
+    @Test
+    void throw_IllegalArgumentException_when_size_is_not_6() {
+        assertThatIllegalArgumentException().isThrownBy(() -> new LottoNumbers(List.of(1, 2, 3, 4, 5)))
+                .withMessage("로또는 6개의 로또 번호로 구성되어야 합니다.");
+    }
+
+    @DisplayName("두 개의 LottoNumbers가 주어졌을 때 매칭되는 갯수를 반환한다.")
+    @MethodSource("generateLottoNumbers")
+    @ParameterizedTest(name = "LottoNumbers: {0}, matchingCount: {1}개")
+    void return_counting_of_matching_lottoNumbers(List<Integer> given, int result) {
+
+        LottoNumbers winningLottoNumbers = new LottoNumbers(List.of(1, 2, 3, 4, 5, 6));
+
+        assertThat(winningLottoNumbers.countMatchingNumbers(new LottoNumbers(given))).isEqualTo(result);
```

Comment:
```suggestion
        var actual = winningLottoNumbers.countMatchingNumbers(new LottoNumbers(given))
        assertThat(actual).isEqualTo(result);
```
어떤 행위를 검증하느냐는 Assertion에 있기보다는 변수로 분리하는 것이 오히려 더 가독성을 올려줄 수 있어요 😄